src/ripple/protocol/impl/TxFormats.cpp

```diff
@@ -394,6 +394,22 @@ TxFormats::TxFormats()
             {sfTicketSequence, soeOPTIONAL},
         },
         commonFields);
+
+    add(jss::DIDSet,
+        ttDID_SET,
+        {
+            {sfURI, soeOPTIONAL},
+            {sfData, soeOPTIONAL},
+            {sfTicketSequence, soeOPTIONAL},
+        },
+        commonFields);
+
+    add(jss::DIDDelete,
+        ttDID_DELETE,
+        {
+            {sfTicketSequence, soeOPTIONAL},
+        },
+        commonFields);
```

Comment:
Similar to my other comment, why not call these something more generic instead of `DIDSet` and `DIDDelete`?

Reply:
Something related to whatever we name the object. If we name the object `DataBlob` (not saying that's a good name), we could call these `DataBlobSet` and `DataBlobDelete`

Reply:
I like the `Collection/Document` nomenclature. I started in NOSQL and always liked those names. If the account is the collection then these could be "Documents".

I agree that DID is more the type of `Data/Document` that is being stored and we should aim for more agnostic approach with the name of the transactor.

Reply:
I was working on my own amendment to do Collections/Documents and one thing I was doing was adding a log functionally.

So when you updated the document, you create a replica and save that index onto the current object. In my code I use `oldDocumentRef` and `newDocumentRef`. This allows me to traverse the logs. 

Technically something like this might be possible from the transaction meta, but I was going to save directly in the object. It was just an experiment for fun so I'm not saying its the right way. If the meta exists then thats probably how it should be done.

I can share the full function but it does involve saving an extra 2 documents on each update.

Its probably outside the scope of this amendment as well, just sharing some information.

Reply:
+1 for `Document`

Reply:
https://github.com/XRPLF/XRPL-Standards/discussions/132